app/src/main/java/de/danoeh/antennapod/activity/SplashActivity.java

```diff
@@ -38,14 +38,14 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             // Trigger schema updates
             PodDBAdapter.getInstance().open();
             PodDBAdapter.getInstance().close();
-            subscriber.onCompleted();
+            subscriber.onComplete();
         })
-            .subscribeOn(Schedulers.newThread())
+            .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe(() -> {
                 Intent intent = new Intent(SplashActivity.this, MainActivity.class);
-                startActivity(intent);
                 finish();
+                startActivity(intent);
```

Comment:
I don't agree with this line swap. This makes the splash screen useless. The splash screen was added to prevent seeing the "non-ready" MainActivity. If the splash is closed before showing MainActivity, the splash does nothing in most cases. The only thing that it does now is flash the screen blue for a few milliseconds. After that, you see MainActivity just as it was before adding the splash.